shotover-proxy/src/transforms/redis/cache.rs

```diff
@@ -89,12 +94,29 @@ impl SimpleRedisCache {
             message.modified = true;
         }
 
-        self.cache_chain
+        let mut messages = self
```

Comment:
This errors if any message is an error?

Just checking with regards to making sure the `stream_id` always match up